evalml/pipelines/components/transformers/transformer.py

```diff
@@ -73,3 +75,11 @@ def fit_transform(self, X, y=None):
 
     def _get_feature_provenance(self):
         return {}
+
+
+class TargetTransformer(Transformer):
```

Comment:
I think we need this abstraction to make it clear which components we expect to be able to invert. It also makes the implementation of `ComponentGraph.from_list` much easier (imo) without having to make every component return both X and y. 

I